examples/data_preprocessing/mimic_dataset/prepare_mimic_cxr.ipynb

```diff
@@ -0,0 +1,1520 @@
+    "rules['rule_id'] = range(len(rules))\n",
+    "rules['label'] = np.concatenate([np.repeat(mentions['label'][i], len(mentions['rule'][i])) for i in range(14)])\n",
+    "rules['label_id'] = [labels2ids[rules['label'][i]] for i in range(len(rules))]\n",
+    "rules.head()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 23,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "T matrix: [[1. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0.]\n",
+      " [1. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0.]\n",
+      " [0. 1. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0.]\n",
+      " [0. 1. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0.]\n",
+      " [0. 1. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0. 0.]]\n",
+      "shape: (148, 14)\n"
+     ]
+    }
+   ],
+   "source": [
+    "rule2rule_id = dict(zip(rules[\"rule\"], rules[\"rule_id\"]))\n",
+    "rule2label = dict(zip(rules[\"rule_id\"], rules[\"label_id\"]))\n",
+    "\n",
+    "def get_mapping_rules_labels_t(rule2label: Dict, num_classes: int) -> np.ndarray:\n",
+    "    \"\"\" Function calculates t matrix (rules x labels) using the known correspondence of relations to decision rules \"\"\"\n",
+    "    mapping_rules_labels_t = np.zeros([len(rule2label), num_classes])\n",
+    "    for rule, labels in rule2label.items():\n",
+    "        mapping_rules_labels_t[rule, labels] = 1\n",
+    "    return mapping_rules_labels_t\n",
+    "\n",
+    "mapping_rules_labels_t = get_mapping_rules_labels_t(rule2label, len(labels2ids))\n",
+    "print(\"T matrix:\", mapping_rules_labels_t[0:5,:])\n",
+    "print(\"shape:\", mapping_rules_labels_t.shape)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We derive our T matrix with 148 rules for 14 classes. "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Now we want to check if there are any rules which assign to the same class. Indeed, there is one case: rule \"defib\" builds a rule for two different classes. In our T matrix, \"defib\" occures in two rows, since knodle cannot handle one rule assigned to multiple classes. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 24,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "False\n",
+      "rule\n",
+      "defib    2\n",
+      "dtype: int64\n"
+     ]
+    }
+   ],
+   "source": [
+    "print(len(np.unique(rules['rule'])) == len(rules['rule']))\n",
+    "rules_size = rules.groupby('rule').size()\n",
+    "print(rules_size[np.where(rules_size > 1)[0]])\n",
+    "#rule defib appears for two different classes"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Z Matrix\n",
+    "The Z matrix contains the information wheter a rule matches to an observation or not. Its shape is #instances x #rules. "
```

Comment:
```suggestion
    "The Z matrix contains the information about the rules matched to the reports. Its shape is #instances x #rules. "
```

I would avoid "observation" term in order to avoid confusion and use words "rule" or "keywords" instead.